### PR TITLE
Remove deprecated CompactionStrategy for version 3.0

### LINKED ISSUE #3111: Remove old Compaction Strategy
[CompactionStrategy](https://github.com/apache/accumulo/blob/8a50c022816086d31a31bd1389ed3246d622d82f/server/tserver/src/main/java/org/apache/accumulo/tserver/compaction/CompactionStrategy.java) was deprecated in version 2.1.0 in favor of `CompactionSelector` and  `CompactionConfigurer` and should be removed for version 3.0.

### PR DESCRIPTION
**Pull request by cshannon**
This removes all of the old CompactionStrategy related classes,tests, and properties.

This closes #3111

### CODE REVIEW

**Comment by cshannon** [comment]
Full IT build finished and passed.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java [hunk 1]
```diff
@@ -383,50 +307,33 @@ public Optional<SortedKeyValueIterator<Key,Value>> getSample(CompactableFile fil
...
     }
 
     @Override
     public Map<String,String> getConfigOverrides(Set<CompactableFile> files) {
-      return computeOverrides(wp);
+      return null;
     }
 
     @Override
     public Set<StoredTabletFile> getFilesToDrop() {
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java:328]: This method should be completely removed from the code as this functionality of dropping files was unique to the compaction strategy and is not supported by the compaction selector.

**Comment by cshannon** [comment]
@keith-turner - Thanks for the feedback, I went through the comments today and I will start addressing them tomorrow.

File: test/src/main/java/org/apache/accumulo/test/functional/SummaryIT.java [hunk 2]
```diff
@@ -518,15 +476,6 @@ public void compactionSelectorTest() throws Exception {
     compactionTest(compactConfig);
```
[the remark this answers is not in the document]
Comment (cshannon) [reply test/src/main/java/org/apache/accumulo/test/functional/SummaryIT.java]: Done in my latest update

File: test/src/main/java/org/apache/accumulo/test/compaction/UserCompactionStrategyIT.java [hunk 3]
```diff
@@ -1,334 +0,0 @@
...
-      c.tableOperations().compact(tableName,
-          new CompactionConfig().setWait(true).setCompactionStrategy(csConfig));
-
-      assertEquals(1, FunctionalTestUtils.countRFiles(c, tableName));
-    }
-
-  }
-
-  @Test
-  public void testConcurrent() throws Exception {
```
[the remark this answers is not in the document]
Comment (cshannon) [reply test/src/main/java/org/apache/accumulo/test/compaction/UserCompactionStrategyIT.java:269]: I added this test to CompactionIT as it seemed like the best place to go.

File: test/src/main/java/org/apache/accumulo/test/compaction/UserCompactionStrategyIT.java [hunk 4]
```diff
@@ -1,334 +0,0 @@
...
-      assertEquals(Set.of("a", "b"), getRows(c, tableName));
-    }
-  }
-
-  @Test
-  public void testDropNone() throws Exception {
-    // test a compaction strategy that selects no files. In this case there is no work to do, want
-    // to ensure it does not hang.
-
-    testDropNone(Map.of("inputPrefix", "Z"));
```
[the remark this answers is not in the document]
Comment (cshannon) [reply test/src/main/java/org/apache/accumulo/test/compaction/UserCompactionStrategyIT.java:132]: I adapted the test from UserCompactionStrategyIT and I added it to CompactionIT called [testSelectNoFiles() ](https://github.com/apache/accumulo/pull/3112/files#diff-677d3e61e8878b63f21837e5ee5f9c0e4422cb12f8f682086a7a0dba305c1bb6R565)

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java (hunk 1 above)
Comment (cshannon) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java]: `getFilesToDrop()` has been removed as well as places that use it

**Review by keith-turner — APPROVED** [review]
Looks great @cshannon

**Comment by cshannon** [comment]
> Looks great @cshannon

Thanks, I'll go ahead and merge this in.

**Pull request merged**